### PR TITLE
LSI-394: Add shared Claude test quality review workflow

### PR DESCRIPTION
**Pull request by tabuckner**
## Summary

- Adds `claude-test-review.yml` reusable workflow for automated unit test quality reviews
- Uses `claude-code-action` with a focused test-quality persona prompt
- Follows the same infrastructure pattern as `claude-code-review.yml` (AWS OIDC, vault secrets, pinned action version)

## What it does

Consumer repos call this workflow on PRs that touch test files. Claude reviews the changed tests for:
1. Bogus/tautological tests (e.g., `expect(true).toBe(true)`)
2. Ineffective assertions (render without asserting)
3. Over-mocking (testing mocks, not real code)
4. Implementation coupling (asserting on internals)
5. Brittle tests (hardcoded DOM structure, snapshot overuse)
6. Missing error/edge cases
7. Async anti-patterns (missing await/act)
8. Copy-paste duplication (should use test.each)
9. Poor test naming
10. Missing cleanup (leaked timers, subscriptions)

Silent on clean PRs - no comment if no issues found. Max 10 inline comments per review.

## Consumer usage

\`\`\`yaml
# .github/workflows/claude-test-review.yml
name: Claude Test Quality Review

on:
  pull_request:
    types: [opened, synchronize]
    paths:
      - '**/*.spec.ts'
      - '**/*.spec.tsx'
      - '**/*.test.ts'
      - '**/*.test.tsx'
      - '**/__tests__/**'

permissions:
  contents: read
  pull-requests: write
  issues: write
  id-token: write
  actions: read
  checks: read
  statuses: read

jobs:
  claude-test-review:
    if: \${{ !endsWith(github.actor, '[bot]') }}
    uses: liquibase/build-logic/.github/workflows/claude-test-review.yml@main
    secrets: inherit
\`\`\`

## Test plan

- [ ] Merge this PR first, then open consumer PR in liquibase/liquibase-insights
- [ ] Open a test PR with a deliberately weak test - verify bot flags it
- [ ] Open a test PR with well-written tests - verify bot stays silent
- [ ] Open a PR with no test file changes - verify workflow does not trigger

LSI-394

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/claude-test-review.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure AWS credentials for vault access
+        uses: aws-actions/configure-aws-credentials@v5
```
[the remark this answers is not in the document]
Comment (tabuckner) [reply .github/workflows/claude-test-review.yml]: This repo's convention is to use semantic version tags for actions, not pinned SHAs (see PR #474 discussion). Dismissing.

File: .github/workflows/claude-test-review.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        with:
+          role-to-assume: ${{ secrets.LIQUIBASE_VAULT_OIDC_ROLE_ARN }}
+          aws-region: us-east-1
+
+      - name: Get secrets from vault
+        id: vault-secrets
+        uses: aws-actions/aws-secretsmanager-get-secrets@v2
```
[the remark this answers is not in the document]
Comment (tabuckner) [reply .github/workflows/claude-test-review.yml]: Same as above - repo convention is version tags, not pinned SHAs. Dismissing.
